### PR TITLE
add Movie model, implements MovieDao, MovieService.

### PR DESCRIPTION
**Pull request by Alkey**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (vladkuzya) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: В чек листе сказано: 
User try with finally, where you use transaction and try with resources with Read operations

**Review by vladkuzya — COMMENTED** [review]
Хорошая работа)

File: src/main/java/com/dev/cinema/MainApp.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public class MainApp {
+    private static MovieService movieService =
+            (MovieService) Injector.getInstance("com.dev.cinema").getInstance(MovieService.class);
+
+    public static void main(String[] args) {
+        Movie movie = new Movie();
+        movie.setTitle("Star Wars");
+        System.out.println(movieService.add(movie));
+        System.out.println(movieService.getAll());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/MainApp.java]:
```suggestion
      movieService.getAll().forEach(System.out::println);
```

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: let's initialize session object in try block

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            transaction.commit();
+            movie.setId(movieId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie entity", e);
+        } finally {
+            session.close();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: what if session == null?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**